### ISSUE TITLE
How to calculate prize index & multi spin

### ISSUE THREAD

**Issue by leineveber** [IvanAdmaers/react-roulette-pro#4]
Hey, I'm new here, I want to implement this roulette to my website but I can't understand how to calculate prize index. As example: I have 4 goods. My backend calculates random number between these items. And this number should win. How to calculate prize index, if my array has more than 4 times? 

One more question, about multispin. My roulette can be opened a few times, this number is also on backend. As example, 4 times. It means that in prop onPrizeDefined I should setStart again to false unless times === 0. But when roulette stops and start sets to false, roulette switches to first item. It should switch only after user clicks "spin" but not after item is won. Can you please help me with using your package?

**Comment by IvanAdmaers** [comment]
Hello, @leineveber! Thank you for your interest in this package.

There are 2 examples in the [example](https://github.com/IvanAdmaers/react-roulette-pro/tree/main/example) folder. I'll try to explain with these examples but if you need more details just let me know.

1) Let's take a look on the "[Advanced Usage](https://github.com/IvanAdmaers/react-roulette-pro/blob/main/example/src/AdvancedUsage.js)" example.
There are functions: `getPrizeIndex` that calculates a random prize index and `reproductionArray` to full the roulette prize items space.
Using `useEffect` you can calculate the prize index value for each new spin.

2) "[Advanced Usage](https://github.com/IvanAdmaers/react-roulette-pro/blob/main/example/src/AdvancedUsage.js)" example. In my implementation I just set `setSpinning(false)` to allow users click on the spin button. All other actions I do in `handleStart` function.

I think the "[Advanced Usage](https://github.com/IvanAdmaers/react-roulette-pro/blob/main/example/src/AdvancedUsage.js)" example will help you, but if you still have any questions, just ask them here.

**Comment by leineveber** [comment]
Thank you for fast reply. I faced a new problem. Roulette breaks (doesn't starts, even `onPrizeDefined` not calls) if prize index is same as previous prize index. But in real life prize index can be the same as previous several times in a row

**Comment by leineveber** [comment]
To reproduce, please open Advanced usage example and edit `getRandomGoodIndex` function to `() => 4 // as example`

**Comment by IvanAdmaers** [comment]
Thank you for your feedback. I’ll explore why does this happen. For now I think you can solve this by using ‘useState’ for prizes and set a new state of them for every new spin.

I’ll let you know when I get explore about this problem.

**Comment by leineveber** [comment]
I've tried but nothing works, maybe I'm doing something wrong. Could you please explain in more details?

**Comment by leineveber** [comment]
Hmm, well, calling `setPrizeIndex(0)` in `onPrizeDefined` solves the problem a bit. In advanced example it looks ok and I'm not sure why, but on my project it doesn't respin over all roulette items, it just moves a bit on the same prize

**Comment by leineveber** [comment]
But if I will add to `useState` key and set this key to roulette & in `handleStart` func call `setKey(Math.random())` it spins :P

**Comment by leineveber** [comment]
Please, add availability to add your own className to roulette wrapper. And, maybe, it will be more easier, if instead of prizeIndex we will use prizeId

**Comment by IvanAdmaers** [comment]
I glad the problem with prizeIndex has been solved. I’ll keep this moment on mind when implement an example in React Roulette Pro 2.0.

An availability of adding your own className to the roulette will be implement in React Roulette Pro 2.0.

I’ll think about to use prizeId instead of prizeIndex. Thank you for your idea.

Do you have any more ideas or questions?

**Comment by leineveber** [comment]
No, thank you a lot for your help and for your package. I'm looking forward for new version :)

**Comment by IvanAdmaers** [comment]
Thank you for support. Feel free to open a new issue if you'll have any questions.

Cheers!
